compiler/src/main/java/com/squareup/anvil/compiler/Utils.kt

```diff
@@ -125,29 +127,50 @@ internal fun AnnotationDescriptor.getAnnotationValue(key: String): ConstantValue
 
 internal fun AnnotationDescriptor.scope(module: ModuleDescriptor): ClassDescriptor {
   val kClassValue = requireNotNull(getAnnotationValue("scope")) as KClassValue
-  return kClassValue.getType(module)
-      .argumentType()
-      .classDescriptorForType()
+  return kClassValue.argumentType(module).classDescriptorForType()
 }
 
 internal fun AnnotationDescriptor.replaces(module: ModuleDescriptor): List<ClassDescriptor> {
   return (getAnnotationValue("replaces") as? ArrayValue)
       ?.value
       ?.map {
-        it.getType(module)
-            .argumentType()
-            .classDescriptorForType()
+        it.argumentType(module).classDescriptorForType()
       }
       ?: emptyList()
 }
 
+internal fun ConstantValue<*>.argumentType(module: ModuleDescriptor): KotlinType {
+  val argumentType = getType(module).argumentType()
+  if (argumentType !is ErrorType) return argumentType
+
+  // Handle inner classes explicitly. When resolving the Kotlin type of inner class from
+  // dependencies the compiler might fail. It tries to load my.package.Class$Inner and fails
+  // whereas is should load my.package.Class.Inner.
+  val normalClass = this.value
+  if (normalClass !is NormalClass) return argumentType
+
+  val classId = normalClass.value.classId
+
+  return module
+      .findClassAcrossModuleDependencies(
+          classId = ClassId(
+              classId.packageFqName,
+              FqName(classId.relativeClassName.asString().replace('$', '.')),
```

Comment:
Does this clash with normal, non-inner classes that have a `$` in them? Considering something like how AutoValue generates class names: `abstract class $$AutoValue_Card extends Card { ... }`. Or would that be handled by some of the code above?